astropy/utils/masked/tests/test_masked.py

```diff
@@ -881,7 +881,7 @@ def test_mean_where(self, axis):
                 [True, True, True],
             ]
         )
-        where_final = ~self.ma.mask & where
+        where_final = (np.isnan(self.ma) | ~self.ma.mask) & where
```

Comment:
What is the motivation for this change?

Reply:
Please revert; `utils.masked` does not have an implicit assumption that `nan` values should be masked.

Reply:
Moving this to a different PR/effort.